chaincfg/params_test.go

```diff
@@ -33,3 +36,51 @@ func TestMustRegisterPanic(t *testing.T) {
 	// Intentionally try to register duplicate params to force a panic.
 	mustRegister(&MainNetParams)
 }
+
+func TestRegisterHDKeyID(t *testing.T) {
+	t.Parallel()
+
+	// Ref: https://github.com/satoshilabs/slips/blob/master/slip-0132.md
+	hdKeyIDZprv := []byte{0x02, 0xaa, 0x7a, 0x99}
+	hdKeyIDZpub := []byte{0x02, 0xaa, 0x7e, 0xd3}
+
+	if err := RegisterHDKeyID(hdKeyIDZpub, hdKeyIDZprv); err != nil {
+		t.Fatalf("RegisterHDKeyID: expected no error, got %v", err)
+	}
+
+	got, err := HDPrivateKeyToPublicKeyID(hdKeyIDZprv)
+	if err != nil {
+		t.Fatalf("HDPrivateKeyToPublicKeyID: expected no error, got %v", err)
+	}
+
+	if !bytes.Equal(got, hdKeyIDZpub) {
+		t.Fatalf("HDPrivateKeyToPublicKeyID: expected result %v, got %v",
+			hdKeyIDZpub, got)
+	}
+}
+
+func TestInvalidHDKeyID(t *testing.T) {
+	t.Parallel()
+
+	prvValid := []byte{0x02, 0xaa, 0x7a, 0x99}
+	pubValid := []byte{0x02, 0xaa, 0x7e, 0xd3}
+	prvInvalid := []byte{0x00}
+	pubInvalid := []byte{0x00}
+
+	if err := RegisterHDKeyID(pubInvalid, prvValid); err != ErrInvalidHDKeyID {
+		t.Fatalf("RegisterHDKeyID: want err ErrInvalidHDKeyID, got %v", err)
+	}
+
+	if err := RegisterHDKeyID(pubValid, prvInvalid); err != ErrInvalidHDKeyID {
+		t.Fatalf("RegisterHDKeyID: want err ErrInvalidHDKeyID, got %v", err)
+	}
+
+	if err := RegisterHDKeyID(pubInvalid, prvInvalid); err != ErrInvalidHDKeyID {
+		t.Fatalf("RegisterHDKeyID: want err ErrInvalidHDKeyID, got %v", err)
+	}
+
+	// FIXME: The error type should be changed to ErrInvalidHDKeyID.
+	if _, err := HDPrivateKeyToPublicKeyID(prvInvalid); err != ErrUnknownHDKeyID {
+		t.Fatalf("HDPrivateKeyToPublicKeyID: want err ErrUnknownHDKeyID, got %v", err)
+	}
```

Comment:
I guess if people rely on checking the value of the err this could be a breaking change, it makes sense to sem-version this before doing the change for sure, so this sounds good to me